pkg/log/logger.go

```diff
@@ -23,50 +23,63 @@ import (
 	"fmt"
 	"reflect"
 	"sync"
+	"sync/atomic"
 
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
 var once sync.Once
-var logger *zap.Logger
-var config *zap.Config
+var holderValue atomic.Value
```

Comment:
We do not need an atomic here. The logger variable is protected by the `sync.Once`. The logger internally protects itself. We cannot build a logger without a config. Neither must change after creation.
On read we just need to return the global. There is nothing that warrants the overhead of the load and the cast of the atomic value.

Reply:
This is here to allow dynamic reconfiguration of the logger. The overhead is minimal (probably a few machine ops per logged statement).

Reply:
Per offline discussions, reverting all logging changes.